crates/radix-engine-toolkit-uniffi/tests/bindings/example_test.go

```diff
@@ -0,0 +1,162 @@
+	assert(t, manifestSummary.EncounteredEntities[0].AddressString() == addresses.ComponentAddresses.Faucet.AddressString(), "Wrong faucet address at index 0")
+	assert(t, manifestSummary.EncounteredEntities[1].AddressString() == addresses.ResourceAddresses.Xrd.AddressString(), "Wrong resource address at index 1")
+	assert(t, manifestSummary.EncounteredEntities[2].AddressString() == account.AddressString(), "Wrong account address at index 2")
+
+	_, err = transactionManifest.Compile()
+	assert(t, err == nil, "Transaction failed to compile")
+
+	var theader = radix_engine_toolkit_uniffi.TransactionHeader{
+		NetworkId:           networkId,
+		StartEpochInclusive: 1,
+		EndEpochExclusive:   10,
+		Nonce:               1,
+		NotaryPublicKey:     *publicKey,
+		NotaryIsSignatory:   true,
+		TipPercentage:       0,
+	}
+
+	var message = radix_engine_toolkit_uniffi.MessagePlainText{
+		Value: radix_engine_toolkit_uniffi.PlainTextMessage{
+			MimeType: "text/plain",
+			Message: radix_engine_toolkit_uniffi.MessageContentStr{
+				Value: "Somr XRD Transfer",
+			},
+		},
+	}
+
+	var intent = radix_engine_toolkit_uniffi.NewIntent(theader, transactionManifest, message)
+	_, err = intent.Hash()
+	assert(t, err == nil, "Transaction Intent hash failed")
+
+	_, err = intent.Compile()
+	assert(t, err == nil, "Transaction Intent compile failed")
+}
+
+func TestRetNotarizedTransaction(t *testing.T) {
+	var networkId uint8 = 1
+	var account1, privateKey1, publicKey1 = newVirtualAccount(t, networkId)
+	var account2, privateKey2, _ = newVirtualAccount(t, networkId)
+	var addresses = radix_engine_toolkit_uniffi.GetKnownAddresses(networkId)
+
+	var mb = radix_engine_toolkit_uniffi.NewManifestBuilder()
+
+	mb, err := mb.FaucetLockFee()
+	assert(t, err == nil, "FaucetLockFee failed")
+
+	mb, err = mb.AccountWithdraw(account1, addresses.ResourceAddresses.Xrd, radix_engine_toolkit_uniffi.DecimalOne())
+	assert(t, err == nil, "AccountWithdraw failed")
+
+	mb, err = mb.AccountDepositEntireWorktop(account2)
+	assert(t, err == nil, "AccountDepositEntireWorktop failed")
+
+	var transactionManifest = mb.Build(networkId)
+
+	var theader = radix_engine_toolkit_uniffi.TransactionHeader{
+		NetworkId:           networkId,
+		StartEpochInclusive: 1,
+		EndEpochExclusive:   10,
+		Nonce:               1,
+		NotaryPublicKey:     *publicKey1,
+		NotaryIsSignatory:   true,
+		TipPercentage:       0,
+	}
+
+	var message = radix_engine_toolkit_uniffi.MessagePlainText{
+		Value: radix_engine_toolkit_uniffi.PlainTextMessage{
+			MimeType: "text/plain",
+			Message: radix_engine_toolkit_uniffi.MessageContentStr{
+				Value: "Somr XRD Transfer",
+			},
+		},
+	}
+
+	var notarizedTransaction, error = radix_engine_toolkit_uniffi.NewTransactionBuilder().
+		Header(theader).
+		Manifest(transactionManifest).
+		Message(message).
+		SignWithPrivateKey(privateKey1).
+		SignWithPrivateKey(privateKey2).
+		NotarizeWithPrivateKey(privateKey1)
+	assert(t, error == nil, "Notarized Transaction failed")
```

Comment:
Can we do an assertion against the output transaction bytes? You can find a test vector from https://github.com/radixdlt/radixdlt-scrypto/blob/ff21f24952318387803ae720105eec079afe33f3/transaction/src/model/versioned.rs#L80.